### PR TITLE
Lazily clean up temp files from DETR partitioner.

### PR DESCRIPTION
**Pull request by alexaryn**
Tested on my laptop both cases (older and younger than cutoff).

### CODE REVIEW

File: lib/sycamore/sycamore/transforms/detr_partitioner.py [hunk 1]
```diff
@@ -161,6 +172,23 @@ def partition_pdf(
 
             return deformable_layout
 
+    def _cleanup_tmp(self):
+        now = time.time()
+        dir = tempfile.gettempdir()
+        for ent in os.scandir(dir):
+            if ent.name.startswith(self.tmp_prefix):
+                try:
+                    st = ent.stat()
```
Comment (HenryL27) [inline lib/sycamore/sycamore/transforms/detr_partitioner.py:181]: what's ent?

**Review by HenryL27 — APPROVED** [review]
mypy be failing?

File: lib/sycamore/sycamore/transforms/detr_partitioner.py (hunk 1 above)
Comment (alexaryn) [reply lib/sycamore/sycamore/transforms/detr_partitioner.py:181]: Short for dirent (struct dirent in C), which is short for directory entry.  The Python type is os.DirEntry.

Comment (HenryL27) [reply lib/sycamore/sycamore/transforms/detr_partitioner.py:181]: can I make you name the variable `entry` or something that in my head doesn't refer to a talking tree person?

**Comment by alexaryn** [comment]
It looks like Python 3.9 TemporaryDirectory() lacks the ignore_cleanup_errors kwarg.  Should be harmless to include it.

**Pull request merged**